packages/core/src/connectors/alipay/index.ts

```diff
@@ -0,0 +1,245 @@
+}
+
+// eslint-disable-next-line unicorn/prefer-module
+const pathToReadmeFile = path.join(__dirname, 'README.md');
+const readmeContentFallback = 'Please check README.md file directory.';
+export const metadata: ConnectorMetadata = {
+  id: 'alipay',
+  type: ConnectorType.Social,
+  name: {
+    en: 'Sign In with Alipay',
+    'zh-CN': '支付宝登录',
+  },
+  // TODO: add the real logo URL (LOG-1823)
+  logo: './logo.png',
+  description: {
+    en: 'Sign In with Alipay',
+    'zh-CN': '支付宝登录',
+  },
+  readme: existsSync(pathToReadmeFile)
+    ? readFileSync(pathToReadmeFile, 'utf8')
+    : readmeContentFallback,
+};
+
+const alipayConfigGuard = z.object({
+  appId: z.string(),
+  charset: z.nativeEnum(CHAR_SET).default(CHAR_SET.UTF8),
+  privateKey: z.string(),
+  signType: z.enum(['RSA', 'RSA2']),
+});
+
+export type AlipayConfig = z.infer<typeof alipayConfigGuard>;
+
+export const validateConfig: ValidateConfig = async (config: unknown) => {
+  const result = alipayConfigGuard.safeParse(config);
+
+  if (!result.success) {
+    throw new ConnectorError(ConnectorErrorCodes.InvalidConfig, result.error.message);
+  }
+};
+
+// Reference: https://github.com/alipay/alipay-sdk-nodejs-all/blob/10d78e0adc7f310d5b07567ce7e4c13a3f6c768f/lib/util.ts
+export const signingPamameters = (
+  parameters: AlipayConfig & Record<string, string>
+): Record<string, string> => {
+  const { biz_content: bizContent, privateKey, ...signParameters } = parameters;
+
+  if (bizContent) {
+    // eslint-disable-next-line @silverhand/fp/no-mutation
+    signParameters.biz_content = JSON.stringify(snakeCaseKeys(JSON.parse(bizContent)));
+  }
+
+  const decamelizeParameters = snakeCaseKeys(signParameters);
+
+  // eslint-disable-next-line @silverhand/fp/no-mutating-methods
+  const sortedParametersAsString = Object.keys(decamelizeParameters)
+    .sort()
+    .map((key) => {
+      // eslint-disable-next-line @silverhand/fp/no-let
+      let data = decamelizeParameters[key];
+
+      if (Array.prototype.toString.call(data) !== '[object String]') {
+        // eslint-disable-next-line @silverhand/fp/no-mutation
+        data = JSON.stringify(data);
+      }
+
+      // Supported Encodings can be found at https://github.com/ashtuchkin/iconv-lite/wiki/Supported-Encodings
+      // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
+      return data ? `${key}=${iconv.encode(data, signParameters.charset)}` : '';
+    })
+    .filter((keyValueString) => keyValueString)
+    .join('&');
+
+  const sign = crypto
+    .createSign(ALIPAY_ALGORITHM_MAPPING[signParameters.signType])
+    .update(sortedParametersAsString, 'utf8')
+    .sign(privateKey, 'base64');
+
+  // eslint-disable-next-line @silverhand/fp/no-mutating-assign
+  return Object.assign(decamelizeParameters, { sign });
+};
+
+export const getAuthorizationUri: GetAuthorizationUri = async (redirectUri, state) => {
+  assertThat(redirectUriRegEx.test(redirectUri), 'connector.invalid_redirect_uri');
```

Comment:
Can we do the check on `routes`? Seems all social connectors have `redirectUri` and share the same regex.